NEWS

```diff
@@ -2,6 +2,79 @@ systemd System and Service Manager
 
 CHANGES WITH 240 in spe:
 
+        * KERNEL API BREAKAGE I: Please note that Linux 4.12 introduced two new
+          uevents "bind" and "unbind", which break all userspace that is not
+          prepared to handle them (which is actually *most* of the packages
+          that process or monitor kernel devices, in all current
+          versions). It's highly recommended to update to systemd/udev 240 when
+          running kernels 4.12 or newer. Moreover, this change in behaviour
+          requires additional changes in almost all other software installing
+          udev rules files, monitoring or enumerating devices with libudev or
+          otherwise processing uevents. Note that this breakage is not fault of
+          systemd or udev, but exclusively caused by an incompatible kernel
```

Comment:
Yeah, it is though, as you did merge my patch enabling handling these 2 new actions in udev.

Reply:
Well, my assumption was that you tested this. udev is just the userspace wrapper around kernel behaviour, and was less than well maintained in the recent past. I am not sure I am willing to take the blame for this breakage, in particular as the your patch simply changed udev to handle these two events like they were processed in older udev versions that didn't validate kernel action types. So, dunno.

> Just for the record, I object to all of this and the way you propose to handle this. If you do not want bind/unbind, filter them out, do not pass it on to rules, and you will be in exactly the same position as you were before 4.12 kernel. Do not require consumers to change the rukes to "ACTION!="remove". This is a recommendation that will hurt in the long run.
> 
> That is my recommendation for now. Whether you decide to implement it or not is up to you obviously.

i am sorry but your proposal can't work. If we filter out "bind" entirely, then all programs waiting for devices to become usable will never notice when they do, as they will only get the "add" at which they are useless still.

Reply:
I tested it on my set up and my use case. I'll take the blame on udev side if you wish ;)